inventory/src/main/java/nl/teamtwee/bep3/restaurant/inventory/core/application/IngredientsCommandHandler.java

```diff
@@ -80,10 +79,31 @@ public void handle(DeleteIngredient command) {
         this.repository.delete(ingredient);
     }
 
+
+    public List<Ingredient> handle(TakeFromStock command) {
+        List<Ingredient> ingredients = new ArrayList<>();
+
+        for (String name : command.getIngredientAmountMap().keySet())
+            ingredients.add(this.repository.findByProductName(name)
+                    .orElseThrow(() -> new IngredientNotFound(name)));
+
+        for (Ingredient ingredient : ingredients) {
+            for (String name : command.getIngredientAmountMap().keySet())
+                ingredient.takeStock(command.getIngredientAmountMap().get(name));
+        }
+
+        for (Ingredient ingredient : ingredients) {
```

Comment:
Deze mag volgens mij in dezelfde for loop als hierboven.

alsin
`for (Ingredient ingredient : ingredients) {
            for (String name : command.getIngredientAmountMap().keySet()){
                ingredient.takeStock(command.getIngredientAmountMap().get(name));
		}
 	this.publishEventsFor(ingredient);
        this.repository.save(ingredient);
        }`